build.js

```diff
@@ -18,39 +26,44 @@ const patterns = {
 	natures: NaturesFile.Natures
 };
 
+const toCapital = (str) => str[0].toUpperCase() + str.slice(1);
+
 for (const [pattern, elements] of Object.entries(patterns)) {
-	let list = [];
-	for (const element in elements) {
-		list.push(elements[element].name);
-	}
-	syntaxFile = syntaxFile.replace('{{' + pattern + '}}', list.sort().reverse().join('|'));
+	const list = Object.values(elements).map(x => x.name);
+	const replaceTarget = '{{' + pattern + '}}';
+	const replaceValue = list.sort().reverse().join('|');
+
+	syntaxFile = syntaxFile.replace(replaceTarget, replaceValue);
 }
 
 console.log('Data gathered, writing language file.');
-fs.writeFileSync('PokemonTeamSyntax.tmLanguage', plist.build(JSON.parse(syntaxFile)));
-console.log('Done.');
 
+fs.writeFileSync(files.tmLanguage, plist.build(JSON.parse(syntaxFile)));
+
+console.log('Done.');
 console.log('-- Building Completions File');
 console.log('Reading current Completions file');
-let completionsFile = JSON.parse(fs.readFileSync('PokemonTeamSyntax.sublime-completions', 'utf-8'));
+
+let completionsFile = JSON.parse(fs.readFileSync(files.sublimeCompletions, 'utf-8'));
 
 console.log('Updating completions from data');
-completionsFile.completions = [];
-let completion = '';
-let tip = '';
+
 for (const [pattern, elements] of Object.entries(patterns)) {
-	for (const element in elements) {
-		completion = elements[element].name;
-		completion = completion.charAt(0).toUpperCase() + completion.slice(1);
-		tip = pattern.charAt(0).toUpperCase() + pattern.slice(1);
-
-		completionsFile.completions.push({
-			trigger: completion + '\t' + tip,
-			contents: pattern !== 'natures' ? completion : completion + ' Nature'
-		});
-	}
+	Object.values(elements).forEach(({ name }, index) => {
+		const type = toCapital(pattern);
+		let completion = toCapital(name);
+
+		if(pattern === 'natures') completion += ' Nature';
+
+		completionsFile.completions[index] = {
```

Comment:
The usage of `index` here overwrites elements of `completionsFile.completions[]` on each loop, because each pattern resets `index` to zero.

My suggestion is to go back to `[].push`ing or use a `[].reduce`r to keep track of current index, like:

```js
Object.entries(patterns).reduce((index, [pattern, elements]) => {
	Object.values(elements).forEach(({ name }) => {
		const type = toCapital(pattern);
		let completion = toCapital(name);

		if(pattern === 'natures') completion += ' Nature';

		completionsFile.completions[index] = {
			trigger: completion + '\t' + type,
			contents: completion
		};

		index++;
	})

	return index;
}, 0);
```

Both ways looks good to me.

Reply:
oops, my bad, I was comparing if it was working properly so I was changing code back and forth, accidentaly forgot to re-add index to forEach 😂

Reply:
Oops, that's right, my bad 😅
But comparing, using reduce here seems like a bazooka on an ant, I think we should revert to the previous array solution you were using.
Also great eyes, I didn't catched that haha